python/taichi/lang/_ndarray.py

```diff
@@ -89,34 +89,41 @@ def __getitem__(self, key):
         """
         raise NotImplementedError()
 
-    @python_scope
-    def fill(self, val):
+    def ndarray_fill(self, val):
         """Fills ndarray with a specific scalar value.
 
         Args:
             val (Union[int, float]): Value to fill.
         """
         if impl.current_cfg().ndarray_use_torch:
             self.arr.fill_(val)
-        else:
+        elif len(self.element_shape) == 0:
             taichi.lang.meta.fill_ndarray(self, val)
+        else:
+            taichi.lang.meta.fill_ndarray_matrix(self, val)
 
-    @python_scope
-    def to_numpy(self):
+    def ndarray_to_numpy(self):
         """Converts ndarray to a numpy array.
 
         Returns:
             numpy.ndarray: The result numpy array.
         """
         if impl.current_cfg().ndarray_use_torch:
             return self.arr.cpu().numpy()
+
         arr = np.zeros(shape=self.arr.shape, dtype=to_numpy_type(self.dtype))
-        taichi.lang.meta.ndarray_to_ext_arr(self, arr)
+        if len(self.element_shape) == 0:
```

Comment:
nit: avoid manually checking this in the parent class

Reply:
@strongoier fixed!